src/main/java/org/folio/note/NoteRepositoryImpl.java

```diff
@@ -0,0 +1,205 @@
+package org.folio.note;
+
+import static io.vertx.core.Future.failedFuture;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotFoundException;
+
+import org.folio.db.model.NoteView;
+import org.folio.rest.jaxrs.model.Note;
+import org.folio.rest.jaxrs.model.NoteCollection;
+import org.folio.rest.persist.PgExceptionUtil;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.persist.Criteria.Limit;
+import org.folio.rest.persist.Criteria.Offset;
+import org.folio.rest.persist.cql.CQLWrapper;
+import org.folio.rest.persist.interfaces.Results;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.z3950.zing.cql.cql2pgjson.CQL2PgJSON;
+import org.z3950.zing.cql.cql2pgjson.CQL2PgJSONException;
+import org.z3950.zing.cql.cql2pgjson.FieldException;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.sql.UpdateResult;
+
+@Component
+public class NoteRepositoryImpl implements NoteRepository {
+
+  private static final String NOTE_VIEW = "note_view";
+  private static final String NOTE_TABLE = "note_data";
+
+  private final Logger logger = LoggerFactory.getLogger(NoteServiceImpl.class);
+
+  @Autowired
+  private Vertx vertx;
+
+  @Override
+  public Future<NoteCollection> getNotes(String cqlQuery, int offset, int limit, String tenantId) {
+    logger.debug("Getting notes. new query:" + cqlQuery);
+    Future<NoteCollection> future = Future.succeededFuture(null);
+    return future.compose(o -> {
+      CQLWrapper cql;
+      try {
+        cql = getCQLForNoteView(cqlQuery, limit, offset);
```

Comment:
there is a utility method in folio-service-tools to get cql wrapper:
https://github.com/folio-org/folio-service-tools/blob/ab98a82c24dd1af8752a72adb6feeb990cd744a6/folio-service-tools-dev/src/main/java/org/folio/db/DbUtils.java#L19